askama_derive/src/input.rs

```diff
@@ -119,8 +119,11 @@ impl TemplateInput<'_> {
             while let Some(nodes) = nested.pop() {
                 for n in nodes {
                     let mut add_to_check = |path: PathBuf| -> Result<(), CompileError> {
-                        let source = get_template_source(&path)?;
-                        check.push((path, source));
+                        if !map.contains_key(&path) {
+                            map.insert(path.clone(), Parsed::default());
```

Comment:
The `parsed` variable has to be owned and not mutably borrowed from `map` because we are going to insert new entries into it. Inserting a dummy entry prevents re-adding the same file multiple times to `check` before we process it.

Reply:
Worth adding a code comment to explain it then. ;)

Reply:
Should I add it to the commit: "[Improve performance of find_used_templates](https://github.com/djc/askama/pull/915/commits/f752cc5396b311afcb8ed104cb76ccd3408f28d2)", or make a separate one?

Reply:
I added the comment

Reply:
:+1: